x-pack/plugins/security_solution/public/detections/components/host_isolation/isolate.tsx

```diff
@@ -15,24 +15,29 @@ import {
   EndpointIsolateForm,
   EndpointIsolateSuccess,
 } from '../../../common/components/endpoint/host_isolation';
+import { CasesFromAlertsResponse } from '../../containers/detection_engine/alerts/types';
 
 export const IsolateHost = React.memo(
   ({
     endpointId,
     hostName,
     cases,
-    caseIds,
+    casesInfo,
     cancelCallback,
   }: {
     endpointId: string;
     hostName: string;
     cases: ReactNode;
-    caseIds: string[];
+    casesInfo: CasesFromAlertsResponse;
     cancelCallback: () => void;
   }) => {
     const [comment, setComment] = useState('');
     const [isIsolated, setIsIsolated] = useState(false);
 
+    const caseIds: string[] = casesInfo.map((caseInfo): string => {
```

Comment:
I think is best to use the `useMemo` here and not in `caseCount`. The reason is that iterating over an array could be expensive but accessing an array attribute (`length`) is not.